book/src/pieces/intro/list_comprehensions_vs_map.md

```diff
@@ -0,0 +1,73 @@
+# List comprehensions vs map/filter
+
+One of Python's most popular features is its list comprehensions. They are a concise way to create
+lists from existing lists. Rust is more functional than Python, so it has a similar feature called
+`map` and `filter`. Although map and filter functions are availble in Python, they are not as
+commonly used as list comprehensions.
+
+## Python
+
+Consider the following function in which we print a message depending on which persons from
+a list of `Person` objects are born after the year 1995, based on their current age.
+
+```py
+def run7() -> None:
+    """
+    1. List comprehensions
+    """
+    persons = [Person("Issa", 39), Person("Ibrahim", 26)]
+    persons_born_after_1995 = [
+        (person.name, person.age) for person in persons if approx_year_of_birth(person) > 1995
+    ]
+    print(f"Persons born after 1995: {persons_born_after_1995}")
+```
+
+The list comprehension in the above function essentially does the following:
+
+1. Iterate over the list of `Person` objects
+2. Unpack each `Person` tuple into their name and age
+3. For each person, check if their approximate year of birth is greater than 1995
+
+Running the above function via `main.py` gives us the following output:
+
+```bash
+Persons born after 1995: [('Ibrahim', 26)]
+```
+
+## Rust
+
+We can define the below function in Rust, where we print a message depending on which persons from
+a vector of `Person` objects are born after the year 1995, based on their current age.
```

Comment:
Have we mentioned anywhere that vectors in rust are roughly the equivalent of lists in Python. If not is it worth mentioning that somewhere?

Reply:
It's important enough that we should mention that, will do!